Have more modules pass parallel and threaded tests

loganharbour opened the pull request:
Refs #16857

This fixes as many modules as I reasonably could within a short timeframe.

The following WIP recipes can be moved to default testing with this PR:

- WIP Modules Partial Parallel: move to Modules Parallel Test
- WIP Modules Partial Parallel PETSc alt: move to Modules Parallel PETSc alt Test
- WIP Modules Recover: new recipe (probably default on PR and next)

The following recipes will remain addable on PRs to test any further fixes for the missing configurations listed in #16857

- WIP Modules Partial Parallel Remaining
- WIP Modules Partial Parallel PETSc alt Remaining
- WIP Modules Recover Remaining

loganharbour committed:
Get contact to pass -p 2 --distributed-mesh testing

refs #16857

loganharbour committed:
Get external_petsc_solver to pass -p 2 --distributed-mesh testing

refs #16857

loganharbour committed:
Get fluid_properties to pass -p 2 --distributed-mesh testing

refs #16857

loganharbour committed:
Get geochemistry to pass -p 2 --distributed-mesh testing

refs #16857

loganharbour committed:
Get heat_conduction to pass -p 2 --distributed-mesh testing

refs #16857

loganharbour committed:
Get peridynamics to pass -p 2 testing

refs #16857

loganharbour committed:
Get chemical_reactions to pass parallel testing

refs #16857

loganharbour committed:
Get stochastic_tools to pass --distributed-mesh testing

refs #16857

loganharbour committed:
Start on getting  phase_field to pass --distributed-mesh testing

refs #16857

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/676437/) on 3a46037 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/16914/site)

This comment will be updated on new commits.

loganharbour commented:
I did the modules here that I could within a reasonable time frame on the couch. Almost all of the additions are reasonable and either should be obviously skipped or required the disabling of renumbering.

@aeslaughter - Take a look at the stochastic tools commit. There's one sub application example where you guys used an `ElementVariableValue` PP that I switched around. There may be a better way to do this, up to you.

@recuero - Take a look at the contact switch. There's a lot of replicated-only testing changes within the verification tests. Would like an approval on this.

WilkAndy commented:
Wow, what a lot of work, thank you @loganharbour

loganharbour commented:
> Wow, what a lot of work, thank you @loganharbour

Surprisingly not - this is only about ~3 hours. But, glad to contribute :-)

GiudGiud committed:
Move tests to Rhie Chow interpolation to avoid diffs due to pressure oscillations
Exact conservation is expected, but not reported in postprocessor outputs due to #16585
Refs #16918

loganharbour commented:
Once #16917 is in, can rebase this and it will be good to go.

recuero commented:
@loganharbour So, basically, this change only makes us enforce a replicated mesh when we add a new test, is that correct?

loganharbour commented:
> @loganharbour So, basically, this change only makes us enforce a replicated mesh when we add a new test, is that correct?

Once we add contact to distributed testing, it will require you to limit tests to replicated mesh if they do not support distributed mesh. By default, you should make a reasonable attempt to get the tests working with distributed mesh if possible.

What my question should have said is: is forcing those tests to be replicated only a valid choice. I couldn't find a reasonable way to get them to pass with distributed mesh. If you wouldn't mind, go run those tests with `-p 2 --distributed-mesh`, look at the errors and see if they're fixable to work with distributed mesh.

recuero commented:
Ghosting will need to be performed differently to make contact work with distributed meshes, so there is not immediate solution to it. Pinging @bwspenc 

I'd say move forward as this PR is for now.

moosebuild commented:
Job [WIP Modules Recover Partial Remaining](https://civet.inl.gov/job/676615/) on d279ea9 : invalidated by @loganharbour

loganharbour commented:
Okay, thanks for the input @recuero

lindsayad committed:
Implement computeElemValuesFace and computeNeighborValuesFace for FV

We were getting incorrect results in our interface postprocessor because
the FV solutions were not initialized

lindsayad committed:
Pass current_elem_is_fi_elem to Moose::FV::interpolate

lindsayad committed:
Regold after interface value calculation framework fix

lindsayad reviewed (commented):
looks fine

lindsayad committed:
nonsingular test should pass with SOME pc

loganharbour commented:
@aeslaughter this should be good now

aeslaughter commented on modules/navier_stokes/test/tests/postprocessors/gold/diverging_INSFV_quad_xy_noslip.csv:
```diff
@@ -1,3 +1,3 @@
 time,inlet_advected_energy,inlet_mass_constant,inlet_mass_matprop,inlet_mass_variable,inlet_momentum_x,inlet_momentum_y,mid1_advected_energy,mid1_mass,mid1_momentum_x,mid1_momentum_y,mid2_advected_energy,mid2_mass,mid2_momentum_x,mid2_momentum_y,outlet_advected_energy,outlet_mass,outlet_momentum_x,outlet_momentum_y
 0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
-1,-303.22631741677,-1,-1,-1,0,-1,310.91762190643,1,-0.00049524449285384,1.1835461195569,310.1724016237,1,0.055136233385212,0.92892382096507,310.03798983956,1,0.015667474006117,0.83548919055089
```
@lindsayad What do you think of these changes?

loganharbour replied:
These changes come from another PR. They're his commits. They're just added to make this pass. It'll go in before this.

aeslaughter commented on modules/navier_stokes/test/tests/postprocessors/gold/diverging_INSFV_quad_xy_upwind.csv:
```diff
@@ -1,3 +1,3 @@
 time,inlet_advected_energy,inlet_mass_constant,inlet_mass_matprop,inlet_mass_variable,inlet_momentum_x,inlet_momentum_y,mid1_advected_energy,mid1_mass,mid1_momentum_x,mid1_momentum_y,mid2_advected_energy,mid2_mass,mid2_momentum_x,mid2_momentum_y,outlet_advected_energy,outlet_mass,outlet_momentum_x,outlet_momentum_y
 0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
-1,-302,-1,-1,-1,-0.024476040741523,-0.94215502182725,310,1,-0.059947284338798,1.0723266687159,310,1,0.029410484806732,0.78663499152553,310,1,0.0034392597619175,0.76467916759833
```
@lindsayad

loganharbour replied:
See above.
